src/compiler/msvc.rs

```diff
@@ -836,6 +844,44 @@ mod test {
         assert!(!msvc_show_includes);
     }
 
+    #[test]
+    fn test_parse_arguments_external_include() {
+        // Parsing -external:I relies on -experimental:external being parsed
+        // and placed into common_args.
```

Comment:
shouldn't `-external:I` be in preprocessor_args, like `-I`?

Reply:
I was assuming, perhaps wrongly, that the actual analysis phase needed information on external directories to issue warnings properly.

Reply:
You'd need both in the same set of args, presumably, but if `-external:I` is there, surely it will have an impact on preprocessing... and without it preprocessing might fail.

Reply:
But `common_args` are passed as part of preprocessing...?

Reply:
Mmmm I was still under the model where compilation was using the preprocessed output, which it doesn't anymore.... except when doing distributed builds. And common_args is passed in that case. Do we want `-external:I` to be passed there?